docs/PublishingGuide.md

```diff
@@ -0,0 +1,67 @@
+## Publishing your report to GitHub Pages
+
+[GitHub Pages](https://pages.github.com/) enables you to publish your report online for free.
+
+1. Fork publish-RepoSense repository using this [link](https://github.com/RepoSense/publish-RepoSense/fork)
+1. Follow this [guide](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) to generate a `Personal access token` on GitHub for report publishing <br/>
+*Remember to **copy it** and you would **only require** `public_repo` permission*
+1. Sign up and login to [Travis-CI](https://travis-ci.org/)
+1. Go to [your account](https://travis-ci.org/account/repositories), click on **Sync account** to fetch all your repositories into Travis-CI
+1. Go to [your fork in Travis-CI](https://travis-ci.org/search/publish-RepoSense/), under **Current** tab click on **Activate repository**
+1. In the same page, click on **More options** on the right then access **Settings**
+![Travis-CI Dashboard](images/publishingguide-travissetting.jpg "Travis-CI Dashboard")
+1. Under **Environment Variables**, name a variable as `GITHUB_TOKEN` and paste the `Personal access token` to its value field <br/>
+*Ensure that the `Display value in build log` is* **switched off** for security reasons
+![Travis-CI Environment Variable](images/publishingguide-githubtoken.jpg "Travis-CI Environment Variable")
+1. Edit [run.sh](../../../../publish-RepoSense/edit/master/run.sh), [repo-config.csv](../../../edit/master/configs/repo-config.csv) and [author-config.csv](../../../edit/master/configs/author-config.csv) according to your preferences <br/>
+*Read our [User Guide](UserGuide.md#customizing-the-analysis) for more information*
+1. To access your site, go to [the setting of your fork](../../../../publish-RepoSense/settings), under **GitHub Pages** look for `Your site is published at [LINK]`
```

Comment:
Can mention that once we have commit the changes to the `publish-RepoSense` fork, a build will be triggered to run the job/script and publish the site. (or need to manually trigger a build if for some reason we don't need to edit the files)

Also, `the setting of your fork` -> `the settings of your fork in GitHub, under *GitHub Pages section*...`